src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,172 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 1 << 4;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int PLACE_FOR_NULL_KEY = 0;
+    private static int capacity;
+    private static int size;
+    private static int threshold;
+    private static int index;
+    private Node<K, V>[] table;
+    private Node<K, V> currentNode;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+        capacity = DEFAULT_INITIAL_CAPACITY;
+        threshold = (int) (DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR);
+        size = 0;
+    }
+
+    private static class Node<K, V> {
+        private final int hash;
+        private final K key;
+        private V value;
+        private Node<K, V> next;
+
+        public Node(int hash, K key, V value, Node<K, V> next) {
+            this.hash = hash;
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (key == null) {
+            if (getForNullKey(table) == null && ++size > threshold) {
+                resize();
+            }
+            putForNullKey(table, value);
+            return;
+        }
+        Node<K, V> newNode = new Node<>(key.hashCode(), key, value, null);
+        if ((currentNode = getNode(key, table, capacity)) == null
+                && ++size > threshold) {
+            resize();
```

Comment:
why do we need to check this `(currentNode = getNode(key, table, capacity)) == null`?